screens/UserScreens/UserDashboard.jsx

```diff
@@ -1,15 +1,50 @@
 import { Modal, View, Text, ScrollView } from "react-native";
-import CreateAndJoinButtons from "../../components/UserDashboard/CreateAndJoin";
-import ProductionButton from "../../components/UserDashboard/ProductionButton";
+import { useState, useEffect } from "react";
+import { onValue, get, ref, child, set } from "firebase/database";
 import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
 import { faClapperboard, faMasksTheater } from "@fortawesome/free-solid-svg-icons";
-import { mockProductions } from "../../MockData/MockProductions";
-import { useState } from "react";
+import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import CreateAndJoinButtons from "../../components/UserDashboard/CreateAndJoin";
+import ProductionButton from "../../components/UserDashboard/ProductionButton";
 import JoinProductionModal from "../../components/ProductionModals/JoinProductionModal";
 import CreateProductionModal from "../../components/ProductionModals/CreateProductionModal";
 
+//TODO:
+// - order by time joined
+// - order participants by last action/ time joined
+
 export default function UserDashboardScreen({ navigation }) {
-    let [modal, setModal] = useState(null);
+    const [modal, setModal] = useState(null);
+    const [createName, setCreateName] = useState("");
+    const [joinCode, setJoinCode] = useState("");
+    const [productions, setProductions] = useState([]);
+    const auth = firebase_auth;
+    const db = firebase_db;
+
+    useEffect(() => {
+        onValue(ref(db, `users/${auth.currentUser.uid}/productions`), async (userSnapshot) => {
+            if (!userSnapshot.exists()) return;
+            const userData = userSnapshot.val();
+
+            const newProds = await Promise.all(
```

Comment:
Promise all? Where's the other promise?

Reply:
every get in the map is a promise - realised that's why everything was breaking before